src/main/scala/chisel3/util/Reg.scala

```diff
@@ -123,6 +123,96 @@ object ShiftRegister {
     compileOptions:      CompileOptions
   ): T =
     ShiftRegisters(in, n, resetData, en).lastOption.getOrElse(in)
+
+  /** Returns the n-cycle delayed version of the input signal (SyncReadMem-based ShiftRegister implementation).
+    *
+    * @param in input to delay
+    * @param n  number of cycles to delay
+    * @param en enable the shift
+    * @param useDualSRAMPort dual port or single port SRAM based implementation
+    * @param name name of SyncReadMem object
+    */
+  def apply[T <: Data](in: T, n: Int, en: Bool, useDualSRAMPort: Boolean, name: Option[String]): T =
```

Comment:
should this .apply have a more clear name, like `mem` or something?

`ShiftReg.mem(...)` ? Or as your test called it, `ShiftMem`?

Reply:
I would rather leave it more general as it is now, nevertheless if you believe `.mem` or something else is more aligned to your code guidelines there's no problem to change it.

Reply:
All right, please have a look one possible implementation that's renaming the `apply` method to `mem` so that the call is now `ShiftRegister.mem`. How does it seem to you?